Cabal/Distribution/Package/TextClass.hs

```diff
@@ -27,6 +27,18 @@ instance Text Dependency where
              Parse.skipSpaces
              return (Dependency name ver)
 
+instance Text ExeDependency where
+  disp (ExeDependency name exe ver) =
+    (disp name <<>> Disp.text ":" <<>> disp exe) <+> disp ver
+
+  parse = do name <- parse
+             _ <- Parse.char ':'
+             exe <- parse
```

Comment:
OK, so right now you are forced to specify the executable name. That's fine (although our users might complain :)